tesseract/tesseract_planning/tesseract_process_managers/include/tesseract_process_managers/taskflow_generator.h

```diff
@@ -63,21 +105,110 @@ class TaskflowGenerator
   /**
    * @brief Generates unique taskflows for using as a component (e.g. with taskflow.composed_of)
    * @param instruction Instructions passed to this instance of the taskflow
-   * @return Reference to taskflow. TaskflowGenerator should maintain ownership
+   * @param done_cb A user defined callback
+   * @param error_cb A user defined callback
+   * @return A taskflow container which must stay around during execution of taskflow
    */
-  virtual tf::Taskflow& generateTaskflow(ProcessInput instruction,
-                                         std::function<void()> done_cb,
-                                         std::function<void()> error_cb) = 0;
+  virtual TaskflowContainer generateTaskflow(ProcessInput instruction,
+                                             std::function<void()> done_cb,
+                                             std::function<void()> error_cb) = 0;
+
+  //  /**
+  //   * @brief Generate a series of task assigned to the provided taskflow but not connected
+  //   * @details The task generated must be attached to other tasks in the taskflow outside this function.
+  //   * Use only the input and outputs tasks for attaching.
+  //   * @param taskflow The taskflow to use for generating new tasks
+  //   * @param instruction Instructions passed to this instance of the taskflow
+  //   * @param done_cb A user defined callback
+  //   * @param error_cb A user defined callback
+  //   * @return A taskflow container which must stay around during execution of taskflow
+  //   */
+  //  virtual TaskflowContainer generateTaskflow(tf::Taskflow& taskflow,
+  //                                             ProcessInput instruction,
+  //                                             std::function<void()> done_cb,
+  //                                             std::function<void()> error_cb) = 0;
+};
+
+/**
+ * @brief The default success task to be used
+ * @param name The name
+ * @param message A detailed message
+ * @param user_callback A user callback function
+ */
+inline void successTask(ProcessInput /*instruction*/,                   // NOLINT
+                        std::string name,                               // NOLINT
+                        std::string message,                            // NOLINT
+                        std::function<void()> user_callback = nullptr)  // NOLINT
```

Comment:
I wanted to ensure that these are being copied because they are being passed to a task and local variable being passed in by reference go out of scope. I am switching the std::bind for these function to using lambda which may solve the issue.

Reply:
I think this is resolved with the latest commit.